js/lessonfilter.js

```diff
@@ -47,74 +51,287 @@ function applySortFromURI(uri, featureList) {
 
   // Performm new sort
   featureList.sort(sortType, { order: sortOrder });
+
+  // Reset filter results header
+  $('#results-value').text($(this).text().split(' ')[0] + '(' + featureList.update().matchingItems.length + ')' + " ");
+  $('#results-value').css('textTransform', 'uppercase');
 }
 
+function lunrSearch(searchString, idx, corpus, featureList, uri) {
+  /* Function to generate search using lunr
+    - find search results using index
+    - load in original corpus and find relevant lessons
+    - generate new html to show search snippet
+    - use featureList to show lessons with search results (also relevant filters)
+    - display html
+  */
+
+  // Get lessons that contain search string using lunr index
+  const results = idx.search(searchString);
+
+  // Get lessons from corpus that contain the search string
+  let docs = results.filter(result => corpus.some(doc => result.ref === doc.url)).map(result => {
+    let doc = corpus.find(o => o.url === result.ref);
+    return {
+      ...result,
+      ...doc
+    };
+  });
+
+  const BUFFER = 30 // Number of characters to show for kwic-results
+  const MAX_KWIC = 3
+  // Create html to show search results using html mark
+  let elements = []
+  docs.map((doc) => {
+    let elementName = '/' + doc.url.split('/').slice(3).join('/')
```

Comment:
😬  Although we're probably not planning on changing our URLs anytime soon, perhaps add a comment here that this code is based on finding the appropriate relative path for a lesson from the absolute path that the search JSON is returning? (that IS what this is trying to do, yes?)

Reply:
Pretty much! In lesson-describe, we're using the lesson url as a unique identifier but that only produces relative urls (/language/lessons/lesson-name). If we wanted it to be url syntax independent, we could add unique ids to all lessons and use that in generating the corpus and the html. But obviously that would be quite a bit of added grunt work. I'll add a comment here though explaining what's going on.

Reply:
Although I wonder if we could just use the slug for each lesson? That would be url independent... 🤔

Reply:
So slug definitely works instead of using absolute url, but we would need to change the search.json and then the search-index repo to test it. Should I push up a separate PR for the search.json so that we can test all the changes using the slug as id, rather than having to use the absolute url?

Reply:
nvm seems like not all posts have slugs... alright back to just commenting the code